pkg/controlplane/controller/peer_endpoint_controller.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controlplane
```

Comment:
this has to go inside its own folder

`pkg/controlplane/controller/peerendpoint/peer_endpoint_controller.go`

Reply:
ok, let me give you the controller for this, since we don't need all the endpoint logic, we just need to update the key, that is going to be our apiserverID with the IP:port information

Reply:
I think that we only need one controller here https://github.com/kubernetes/kubernetes/pull/117740/files#r1262609260

Reply:
refactored the peer_endpoint_lease.go to contain the controller